libraries/AC_Fence/AC_Fence.cpp

```diff
@@ -470,6 +471,34 @@ bool AC_Fence::check_fence_alt_min()
     return false;
 }
 
+
+/// auto enable fence floor
+bool AC_Fence::auto_enable_fence_floor()
+{
+    if (_floor_enabled) {
+        return true; // already enabled
+    }
+
+    // altitude fence check
+    if (!(_enabled_fences & AC_FENCE_TYPE_ALT_MIN)) {
+        // not enabled
+        return false;
+    }
+
+    AP::ahrs().get_relative_position_D_home(_curr_alt);
```

Comment:
this needs to be a local variable then assign the right value, or we have a race that another thread (eg. scripting) could hit

Reply:
Fixed thanks